### ISSUE TITLE
Challenge 24 - AI energy

### ISSUE THREAD

**Issue by DiegoDomenici** [ECMWFCode4Earth/Challenges_2026#14]
> **Stream 2 - Machine Learning**

### Mentors

- Edward Comyn-Platt
- Chiara Cagnazzo
- Fiona Spuler
- Ben Aslan 

### Skill Required

- Python skills
- Experience with ML algorithms

### Goal

Investigate the possibility of using the causal approach to ML in Earth Science (Spuler et al 2024 and 2025) for Energy sector applications.

<hr>

> <b>  *Note: The funding source for this challenge is ECMWF funding. For details on the eligibility, please refer to Article 3 of the [Terms and Conditions](https://codeforearth.ecmwf.int/terms-and-conditions).* </b>

<hr>

### Description of the Challenge

Spuler et al [2024](https://www.cambridge.org/core/journals/environmental-data-science/article/identifying-probabilistic-weather-regimes-targeted-to-a-localscale-impact-variable/D0F1F80FE4ACD2B85F4651FF69F1ED0B), [2025](https://wcd.copernicus.org/articles/6/995/2025/) introduced a causal machine learning approach to identify teleconnections targeted to an impact of interest and demonstrated that the method is able to more accurately capture predictable drivers of extreme precipitation over Morocco. The approach is based on a variational autoencoder architecture that captures variability in the input space while disentangling the reduced dimension modulating the target variable. The identified drivers can either be represented as probabilistic clusters to learn targeted circulation regimes, or as a continuous index.

This challenge should investigate the feasibility of using this approach for problems relevant to the Energy sector. Previous studies have highlighted the influence large scale weather regimes may have on energy demand and production and its subseasonal predictability [van der Wiel et al 2019](https://iopscience.iop.org/article/10.1088/1748-9326/ab38d3/meta), Rouges et al 2024). However, current approaches to identify drivers of period of low renewable energy production and high demand (e.g. Dunkelflauten) in a data-driven approach have shown to compromise the predictability of these drivers, limiting their use in subseasonal applications (Bloomfield et al 2020, Bloomfield et al 2021). Here, the new approach proposed in Spuler et al 2024 offers promise.
 
The successful challengers should formalise the [existing code](https://zenodo.org/records/17177599) and create an importable python package that can be installed via PyPi. Demonstrative example notebooks of using the software using real meteorology and societal data should be included.

### Evaluation criteria

- Feasibility
- Innovative approach
- Transferability
- Easy to maintain / Future-proof approach
- Comprehensibility
- Matching requirements;

**Comment by fiona511** [comment]
Thank you for your interest in the project! Below are some answers to your questions:
- Ultimately, our interest is in identifying large-scale drivers of electricity demand and renewable energy production; however, we would consider working with meteorological intermediary variables, or proxies of electricity demand and renewable energy production derived from purely meteorological variables, as a viable first step.
- In terms of the main goal, yes, that is correct, although it could be slightly rephrased as creating a package for VAE-based regime-based identification and demonstrating its application to the energy sector. How much emphasis should be put on method comparison and the evaluation framework depends on the users you decide to focus on (i.e., whether you envision users already familiar with weather regimes who are assumed to have their own evaluation pipelines and out-of-the-box methods implemented). From our perspective, incorporating evaluation metrics of the VAE-based regimes themselves would be desirable.
- Our hypothesis, which we state in the paper, is that the regimes maintain predictability because the architecture prioritises reconstruction (i.e., identifying regimes that capture main modes of predictable variability in the input space and thereby minimizes the reconstruction loss term) alongside informativeness, which, in the CMM-VAE method, is enforced through the cluster assignments (c_1, .. ,c_k) - thereby ‘resolving’ areas of the phase space associated with target variable while still capturing overlal predictable variability. We have tested the method in several applications and have found consistent results; however, there is no proof that this should hold in any application, and it would be great to explore this question further.
- VAEs overall can be sensitive to the initial seed set in the training; therefore, developing a robust training framework to identify suitable hyperparameters is an important part of this project. For this method in particular, the less closely related the target variable is to the driver variable, the more unstable the architecture becomes. This links back to your first question on whether training the architecture on meteorological variables or energy variables will produce more robust results, which will be interesting to explore.

Looking forward to your proposal!

**Comment by EsperanzaCuartero** [comment]
Dear [qnicolas](https://github.com/qnicolas)
Could you please check your email inbox? You have a message from the Code for Earth Coordination Team, and we kindly ask you to reply as soon as possible. Thanks in advance.
